env0/resource_environment_import.go

```diff
@@ -0,0 +1,132 @@
+				ValidateDiagFunc: NewStringInValidator([]string{"github", "gitlab", "bitbucket", "azure", "other"}),
+			},
+			"workspace": {
+				Type:        schema.TypeString,
+				Description: "workspace of the environment",
+				Optional:    true,
+			},
+			"iac_type": {
+				Type:             schema.TypeString,
+				Description:      "iac type of the environment ( one of \"opentofu\" | \"terraform\" )",
+				Optional:         true,
+				ValidateDiagFunc: NewStringInValidator([]string{"opentofu", "terraform"}),
+			},
+			"iac_version": {
+				Type:        schema.TypeString,
+				Description: "iac version of the environment",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceEnvironmentImportCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	var payload client.EnvironmentImportCreatePayload
+	if err := readResourceData(&payload, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	if d.Get("git_provider").(string) != "" {
+		if err := readResourceData(&payload.GitConfig, d); err != nil {
+			return diag.Errorf("schema resource data deserialization failed: %v", err)
+		}
+	}
+
+	environmentImport, err := apiClient.EnvironmentImportCreate(&payload)
+	if err != nil {
+		return diag.Errorf("could not create environment import: %v", err)
+	}
+
+	d.SetId(environmentImport.Id)
+
+	return nil
+}
+
+func resourceEnvironmentImportRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	environmentImport, err := apiClient.EnvironmentImportGet(d.Id())
+	if err != nil {
+		return ResourceGetFailure(ctx, "environment_import", d, err)
+	}
+
+	if err := writeResourceData(environmentImport, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	if err := writeResourceData(&environmentImport.GitConfig, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	return nil
+}
+
+func resourceEnvironmentImportUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	id := d.Id()
+	var payload client.EnvironmentImportUpdatePayload
+
+	if err := readResourceData(&payload, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	if err := readResourceData(&payload.GitConfig, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	if _, err := apiClient.EnvironmentImportUpdate(id, &payload); err != nil {
+		return diag.Errorf("could not update environment import: %v", err)
+	}
+
+	return nil
+}
+
+// should not actually delete the environment import
+func resourceEnvironmentImportDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return nil
```

Comment:
I am not sure this is correct. Can you please explain why you want to create a drift?
If it's really required, perhaps add an explanation to the code.